pymc_experimental/model_builder.py

```diff
@@ -187,18 +187,15 @@ def load(cls, fname):
         filepath = Path(str(fname))
         data = az.from_netcdf(filepath)
         idata = data
-        # Since there is an issue with attrs getting saved in netcdf format which will be fixed in future the following part of code is commented
-        # Link of issue -> https://github.com/arviz-devs/arviz/issues/2109
-        # if model.idata.attrs is not None:
-        #     if model.idata.attrs['id'] == self.idata.attrs['id']:
-        #         self = model
-        #         self.idata = data
-        #         return self
-        #     else:
-        #         raise ValueError(
-        #             f"The route '{file}' does not contain an inference data of the same model '{self.__name__}'"
-        #         )
-        return idata
+        if idata.attrs is not None:
+            if self.id() == idata.attrs["id"]:
+                self = cls(idata.attrs["sample_config"], idata.attrs["model_config"])
```

Comment:
No this needs to be outside. First create the object `self`, put in the idata, then make the check for id.

Reply:
This statement is to check if the self and the inference data from `.nc` has same id. If we create the self with the loaded data then how will we check if idata from the file and the current model has same id

Reply:
Yes, but we first need to create the `self` object, we can only check `self.id()` after.

Reply:
You can check by trying to run this.

Reply:
Does this make sense:
```python
        filepath = Path(str(fname))
        data = az.from_netcdf(filepath)
        idata = data
        self = cls(idata.attrs["sample_config"], idata.attrs["model_config"])
        self.idata = idata
        if self.id() != idata.attrs["id"]:
            raise ValueError(
                f"The route '{file}' does not contain an inference data of the same model '{self.__name__}'"
            )

        return self
```